components/script/dom/window.rs

```diff
@@ -2310,7 +2310,10 @@ impl Window {
         self.Document().url()
     }
 
-    pub fn with_layout<T>(&self, call: impl FnOnce(&mut dyn Layout) -> T) -> Result<T, ()> {
+    pub fn with_layout<T>(
+        &self,
+        call: impl FnOnce(&mut dyn Layout) -> T,
+    ) -> Result<T, script_thread::LayoutNotFoundError> {
```

Comment:
I don't think we want to create `LayoutNotFoundError`. Instead, it would be better to ensure that a layout is always found, which should be true. I have been meaning to make this change.